Update the description of Enarx goals in the README (#916)
enarx/enarx: README.md

# Enarx

## Introduction Enarx is an application deployment system enabling applications

> to run within Trusted Execution Environments (TEEs) without rewriting for

Reviewer: Was it your intention to have the first part of this paragraph as a header here?

If so, I think the word "Introduction" should be removed. 

If not, I think the words following "Introduction" should be moved into the paragraph section.

Author: Good catch. Will correct, thanks.

> particular platforms or SDKs. It handles attestation and delivery into a

Reviewer: I think something along these lines would add more precision to this paragraph:

"It handles attestation and delivery of WebAssembly workloads into a run-time "Keep", [..]"

Author: I think that's already too specific, we want to keep this both factually correct and readable by various audiences, including some for which "WebAssembly workload" will be less clear than "based on WebAssembly", which is arguably more hand-wavy, but on purpose.

My feeling:
"WebAssembly workloads" --> Wait, do I use that? Am I doing that? Will I need to change things?
"based on WebAssembly" --> Right, I've heard of that technology, let's see how they use it.

run-time “Keep” based on WebAssembly, offering developers a wide range of
language choices for implementation. Enarx is CPU-architecture independent,
enabling the same application code to be deployed across multiple targets,
abstracting issues such as cross-compilation and differing attestation
mechanisms between hardware vendors. Work is currently underway on AMD SEV and
Intel SGX.

We've known for a long time that we need encryption for data at rest and in

> transit: Enarx helps you do encryption for data as well as algorithms in use.

Reviewer: "data ~~as well as~~ and algorithms in use."

> We've known for a long time that we need encryption for data at rest and in
> transit: Enarx helps you do encryption for data as well as algorithms in use.

Reviewer 2: I don't like "We" and "you".

```suggestion
Enarx ensures to keep data at rest and in transit encrypted, as well as the algorithms in use.
```

[Read more](https://github.com/enarx/enarx/wiki/Enarx-Introduction)

## Project aim

> The aim of the Enarx project is to create a way to allow easy deployment of

Reviewer 3: ```suggestion
Create a way to allow easy deployment of
```

We already specify that this is the project's aim; no need to say it twice.

Author: Don't entirely agree, "Project aim" is the name on the section. I'd argue using fully formed sentences after the title of the section to repeat the same idea is fine, the opposite is a bit too terse.

applications using Trusted Execution Environments (TEEs) across multiple
platforms, with strong security properties.

> Deployed workloads should be:

Reviewer: ~~Deployed workloads should be~~ -> "Enarx-deployed workloads will have these properties:"

Author: Thanks, we want to keep the wording simple, so i think we'll keep the current one.

> * Confidentiality protected from the host and other workloads

Reviewer: Suggestion:

"Confidential: nothing (except for the workload itself) has the visibility required to disclose the workload's sensitive information. All unauthorized access, even from the host computer the workload is deployed on, is blocked."

Reviewer 3: A suggestion to your suggestion: "Confidentiality" in your first sentence should become "Confidential".

Otherwise, it reads like "Deployed workloads should be confidentiality", which isn't right.

Reviewer 3: A suggestion to your suggestion: "Confidentiality" in your first sentence should become "Confidential".

Otherwise, it reads like "Deployed workloads should be confidentiality", which isn't right.

Author: This is good but should probably go in another document, where we can explain this in more detail. We are trying to keep the readme short and simple. @MikeCamel, what do you think?

> * Integrity protected from the host and other workloads

Reviewer: Suggestion:

"Integrity-protected: all attempts to manipulate the workload's memory (whether malicious or by accident) are blocked. The workload's data will _never_ change without its blessing."

Author: Same as previous comment. Good explanation, but probably better suited to a longer intro.

* Cryptographically attested at every instance deployment

> * Platform-agnostic at deployment time

Reviewer: I think it's more effective to remove this bullet point and just pull its sub-bullet points out to the top level

Author: I think we've been using this wording (platform-agnostic, deployment time) and should keep it for now.
